software/23.Elixir/README.md

```diff
@@ -19,4 +19,143 @@ Let's help him build the chat app of his dreams ! :)~
+import "../css/app.css"
+
+// If you want to use Phoenix channels, run `mix help phx.gen.channel`
+// to get started and then uncomment the line below.
+-// import "./user_socket.js"
++ import "./user_socket.js"
+```
+
+We are also prompted to update `endpoint.ex`, to implement our channel. 
+The file is stored in `lib/irc_chat_web/endpoint.ex`.
+
+A `socket /live` statemtent should at line 13, you can put your socket justunder this one.
+
+```elixir
+defmodule IrcChatWeb.Endpoint do
+  use Phoenix.Endpoint, otp_app: :irc_chat
+
+  # The session will be stored in the cookie and signed,
+  # this means its contents can be read but not tampered with.
+  # Set :encryption_salt if you would also like to encrypt it.
+  @session_options [
+    store: :cookie,
+    key: "_irc_chat_key",
+    signing_salt: "1jRoL3v2"
+  ]
+
+  socket "/live", Phoenix.LiveView.Socket, websocket: [connect_info: [session: @session_options]]
+  socket "/socket", IrcChatWeb.UserSocket, websocket: true, longpoll: false
+
+...
+```
+
+## Step 3 -- Build the Socket Handler
+
+Next, we will need to create the piece of code to handle our socket. You can create a new file in `/lib/irc_chat_web`. It will define the `UserSocket` module, so I would advise you to name it `user_socket.ex`.
+
+Socket params are passed from the client and can be used to verify and authenticate a user. 
+You can find some apt documentation [here](https://hexdocs.pm/phoenix/Phoenix.Socket.html).
+
+> Dont forget to add a `channel` for your chat room ! 
+> (Google is your friend if you don't know what that means)
+
+## Step 4 -- Add a chat form to the front-end
+Phoenix bundles [Miligram CSS](https://milligram.io/grids.html) by default, so you can use this to build your form ! 
+To handle front-end developpement, Phoenix (Elixir's Web Framework) uses a [Templating engine](https://www.educative.io/edpresso/what-are-template-engines). 
+
+Here is what I added to my template file to have a functionnal form:
+```html
+<ul id="msg-list" style="list-style: none; min-height:200px;">
+</ul>
+
+<div class="row">
+  <div class="col-xs-3" style="width: 20%; margin-left: 0;">
+    <input type="text" id="name" class="form-control" placeholder="Your Name" style="border: 1px black solid; font-size: 1.3em;" autofocus>
+  </div>
+  <div class="col-xs-9" style="width: 100%; margin-left: 1%; ">
+    <input type="text" id="msg" class="form-control" placeholder="Your Message" style="border: 1px black solid; font-size: 1.3em;">
+  </div>
+</div>
+```
+
+This should be added in a file finishing by `.eex`.
+To know what it represents, this should be added in a specific flle: `/lib/irc_chat_web/templates/page/index.html.eex`
+
+Try adding your own form to your template file, and make it prettier if you feel like it...
+After that, you will need to update your `lib/irc_chat_web/templates/layout/app.html.eex` and update the `<header>` tags to handle the newly added code.
```

Comment:
I thing it's could be intersting to list steps and show an example of a final result here too.